src/test/deploy/functions/cache/applyHash.spec.ts

```diff
@@ -0,0 +1,117 @@
+import * as hash from "../../../../deploy/functions/cache/hash";
+import { applyBackendHashToBackends } from "../../../../deploy/functions/cache/applyHash";
+import * as backend from "../../../../deploy/functions/backend";
+import { expect } from "chai";
+import * as sinon from "sinon";
+
+const EMPTY_ENDPOINT: backend.Endpoint = {
+  id: "id",
+  region: "region",
+  project: "project",
+  platform: "gcfv2",
+  runtime: "nodejs16",
+  entryPoint: "ep",
+  httpsTrigger: {},
+  secretEnvironmentVariables: [],
+};
+
+describe("applyHash", () => {
+  afterEach(() => {
+    sinon.verifyAndRestore();
+  });
+
+  describe("applyBackendHashToBackends", () => {
+    it("should applyHash to each endpoint of a given backend", async () => {
+      // Prepare
+      const context = {
+        projectId: "projectId",
+        sources: {
+          backend1: {
+            functionsSourceV1: "backend1_sourceV1",
+            functionsSourceV2: "backend1_sourceV2",
+          },
+          backend2: {
+            functionsSourceV1: "backend2_sourceV1",
+            functionsSourceV2: "backend2_sourceV2",
+          },
+        },
+      };
+      const endpoint1 = {
+        ...EMPTY_ENDPOINT,
+        id: "endpoint1",
+        platform: "gcfv1",
+        codebase: "backend1",
+        secretEnvironmentVariables: [
+          {
+            key: "key",
+            secret: "secret1",
+            projectId: "projectId",
+            version: "1",
+          },
+        ],
+      } as backend.Endpoint;
```

Comment:
nit* I'd try to avoid type assertion. It make it really hard to refactor.

Prefer `const endpoint1: backend.Endpoint` if appropriate,

Reply:
Updated!